gprofiler/profilers/profiler_base.py

```diff
@@ -120,22 +121,31 @@ def snapshot(self) -> ProcessToStackSampleCounters:
         with concurrent.futures.ThreadPoolExecutor(max_workers=len(processes_to_profile)) as executor:
             futures = {}
             for process in processes_to_profile:
-                futures[executor.submit(self._profile_process, process)] = process.pid
+                try:
+                    comm = process_comm(process)
+                except NoSuchProcess:
+                    continue
+
+                futures[executor.submit(self._profile_process, process)] = (process.pid, comm)
 
             results = {}
             for future in concurrent.futures.as_completed(futures):
+                pid, comm = futures[future]
                 try:
                     result = future.result()
-                    if result is not None:
-                        results[futures[future]] = result
+                    assert result is not None
                 except StopEventSetException:
                     raise
                 except NoSuchProcess:
                     logger.debug(
-                        f"{self.__class__.__name__}: process went down during profiling {futures[future]}",
+                        f"{self.__class__.__name__}: process went down during profiling {pid} ({comm})",
                         exc_info=True,
                     )
-                except Exception:
-                    logger.exception(f"{self.__class__.__name__}: failed to profile process {futures[future]}")
+                    result = self._profiling_error_stack("error", "process went down during profiling", comm)
+                except Exception as e:
+                    logger.exception(f"{self.__class__.__name__}: failed to profile process {pid} ({comm})")
+                    result = self._profiling_error_stack("error", str(e), comm)
```

Comment:
`str(e)` might be too much, maybe `type(e).__name__` is enough?

Reply:
Hmm I suppose we can't know in advance, I'll fix